{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Microbiota analysis\n",
    "## Author: Tijs van Lieshout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 188 entries, 0 to 187\n",
      "Data columns (total 16 columns):\n",
      " #   Column                                    Non-Null Count  Dtype  \n",
      "---  ------                                    --------------  -----  \n",
      " 0   Unique ID                                 188 non-null    object \n",
      " 1   UP ID                                     188 non-null    object \n",
      " 2   UP taxonomy                               188 non-null    int64  \n",
      " 3   UP name                                   188 non-null    object \n",
      " 4   Assembly ID (UP matched)                  188 non-null    object \n",
      " 5   RefSeq Accession                          188 non-null    object \n",
      " 6   Plasmid/additional chromosome accessions  44 non-null     object \n",
      " 7   GenBank Accession                         188 non-null    object \n",
      " 8   GenBank record name                       188 non-null    object \n",
      " 9   NCBI taxonomy                             187 non-null    float64\n",
      " 10  NCBI taxonomy name                        186 non-null    object \n",
      " 11  Lineage (full)                            188 non-null    object \n",
      " 12  Characteristics                           183 non-null    object \n",
      " 13  Effect/Function                           180 non-null    object \n",
      " 14  Curation Notes                            135 non-null    object \n",
      " 15  Genome Size (Mb)                          185 non-null    float64\n",
      "dtypes: float64(2), int64(1), object(13)\n",
      "memory usage: 23.6+ KB\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 67 entries, 0 to 66\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          67 non-null     object \n",
      " 1   NCBI_tax_id         67 non-null     object \n",
      " 2   relative_abundance  67 non-null     float64\n",
      " 3   additional_species  11 non-null     object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 2.2+ KB\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1 entries, 0 to 0\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          1 non-null      object \n",
      " 1   NCBI_tax_id         1 non-null      int64  \n",
      " 2   relative_abundance  1 non-null      float64\n",
      " 3   additional_species  0 non-null      float64\n",
      "dtypes: float64(2), int64(1), object(1)\n",
      "memory usage: 160.0+ bytes\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1 entries, 0 to 0\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          1 non-null      object \n",
      " 1   NCBI_tax_id         1 non-null      int64  \n",
      " 2   relative_abundance  1 non-null      float64\n",
      " 3   additional_species  0 non-null      float64\n",
      "dtypes: float64(2), int64(1), object(1)\n",
      "memory usage: 160.0+ bytes\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 51 entries, 0 to 50\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          51 non-null     object \n",
      " 1   NCBI_tax_id         51 non-null     object \n",
      " 2   relative_abundance  51 non-null     float64\n",
      " 3   additional_species  13 non-null     object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 1.7+ KB\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 9 entries, 0 to 8\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          9 non-null      object \n",
      " 1   NCBI_tax_id         9 non-null      object \n",
      " 2   relative_abundance  9 non-null      float64\n",
      " 3   additional_species  1 non-null      object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 416.0+ bytes\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 104 entries, 0 to 103\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          104 non-null    object \n",
      " 1   NCBI_tax_id         104 non-null    object \n",
      " 2   relative_abundance  104 non-null    float64\n",
      " 3   additional_species  20 non-null     object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 3.4+ KB\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 29 entries, 0 to 28\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          29 non-null     object \n",
      " 1   NCBI_tax_id         29 non-null     object \n",
      " 2   relative_abundance  29 non-null     float64\n",
      " 3   additional_species  3 non-null      object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 1.0+ KB\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 9 entries, 0 to 8\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          9 non-null      object \n",
      " 1   NCBI_tax_id         9 non-null      object \n",
      " 2   relative_abundance  9 non-null      float64\n",
      " 3   additional_species  1 non-null      object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 416.0+ bytes\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 24 entries, 0 to 23\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          24 non-null     object \n",
      " 1   NCBI_tax_id         24 non-null     object \n",
      " 2   relative_abundance  24 non-null     float64\n",
      " 3   additional_species  7 non-null      object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 896.0+ bytes\n",
      "None\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 41 entries, 0 to 40\n",
      "Data columns (total 4 columns):\n",
      " #   Column              Non-Null Count  Dtype  \n",
      "---  ------              --------------  -----  \n",
      " 0   clade_name          41 non-null     object \n",
      " 1   NCBI_tax_id         41 non-null     object \n",
      " 2   relative_abundance  41 non-null     float64\n",
      " 3   additional_species  9 non-null      object \n",
      "dtypes: float64(1), object(3)\n",
      "memory usage: 1.4+ KB\n",
      "None\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "PATH = \"../microbiota_tax_data\"\n",
    "gfkb = pd.read_csv(f\"{PATH}/GutFeelingKnowledgeBase-v4-Master_List.csv\")\n",
    "gfkb = gfkb.drop(columns=[\"Present in GFKB v3 (Y/N)\", \"Present in GFKB_epilepsy v3 (Y/N)\"])\n",
    "gfkb = pd.concat([gfkb.drop(columns=[\"Genome Size (Mb)\"]), gfkb[\"Genome Size (Mb)\"]], axis=1)\n",
    "\n",
    "gfkb.info()\n",
    "\n",
    "tax_profiles = []\n",
    "for file in glob.glob(f\"{PATH}/metaphlan_output/*.txt\"):\n",
    "    tax_profile = pd.read_csv(file, \n",
    "                              comment=\"#\", \n",
    "                              sep=\"\\t\", \n",
    "                              names=[\"clade_name\", \"NCBI_tax_id\", \"relative_abundance\", \"additional_species\"])\n",
    "    tax_profiles.append(tax_profile)\n",
    "    print(tax_profile.info())\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
